test/distributed/_shard/test_replicated_tensor.py

```diff
@@ -74,3 +76,109 @@ def test_replicated_tensor_inter_op_tensor(self):
+            and not isinstance(new_tensor, ReplicatedTensor)
+        )
+        self.assertEqual(new_tensor.local_tensor(), st.local_tensor() + 4)
+        # test torch.add
+        new_tensor = torch.add(replica_tensor, st)
+        self.assertEqual(new_tensor.local_tensor(), st.local_tensor() + 4)
+
+        new_tensor = st + replica_tensor
+        self.assertEqual(new_tensor.local_tensor(), st.local_tensor() + 4)
+
+        # sub op
+        new_tensor = replica_tensor - st
+        self.assertTrue(
+            isinstance(new_tensor, sharded_tensor.ShardedTensor)
+            and not isinstance(new_tensor, ReplicatedTensor)
+        )
+        self.assertEqual(new_tensor.local_tensor(), 4 - st.local_tensor())
+        # test torch.sub
+        new_tensor = torch.sub(replica_tensor, st)
+        self.assertEqual(new_tensor.local_tensor(), 4 - st.local_tensor())
+
+        new_tensor = st - replica_tensor
+        self.assertEqual(new_tensor.local_tensor(), st.local_tensor() - 4)
+
+        # mul op
+        new_tensor = replica_tensor * st
+        self.assertTrue(
+            isinstance(new_tensor, sharded_tensor.ShardedTensor)
+            and not isinstance(new_tensor, ReplicatedTensor)
+        )
+        self.assertEqual(new_tensor.local_tensor(), 4 * st.local_tensor())
+        # test torch.mul
+        new_tensor = torch.mul(replica_tensor, st)
+        self.assertEqual(new_tensor.local_tensor(), 4 * st.local_tensor())
+
+        new_tensor = st * replica_tensor
+        self.assertEqual(new_tensor.local_tensor(), 4 * st.local_tensor())
+
+        # div op
+        new_tensor = replica_tensor / st
+        self.assertTrue(
+            isinstance(new_tensor, sharded_tensor.ShardedTensor)
+            and not isinstance(new_tensor, ReplicatedTensor)
+        )
+        self.assertEqual(new_tensor.local_tensor(), 4 / st.local_tensor())
+        # test torch.div
+        new_tensor = torch.div(replica_tensor, st)
+        self.assertEqual(new_tensor.local_tensor(), 4 / st.local_tensor())
+
+        new_tensor = st / replica_tensor
+        self.assertEqual(new_tensor.local_tensor(), st.local_tensor() / 4)
+
+    @with_comms(init_rpc=False)
+    @skip_if_lt_x_gpu(4)
+    @requires_nccl()
+    def test_replicated_tensor_inter_op_sharded_tensor_errors(self):
+        local_tensor = torch.ones(3, 3, device=f"cuda:{self.rank}") * 4
+        replica_tensor = ReplicatedTensor(local_tensor)
+
+        torch.manual_seed(self.rank)
+        spec = ChunkShardingSpec(
+            dim=0,
+            placements=[
+                "rank:0/cuda:0",
+                "rank:1/cuda:1",
+                "rank:2/cuda:2",
+                "rank:3/cuda:3",
+            ],
+        )
+
+        st = sharded_tensor.rand(spec, (20, 3))
+
+        with self.assertRaisesRegex(RuntimeError, 'Implicit broadcasting'):
+            st + replica_tensor
+
+        with self.assertRaisesRegex(TypeError, 'unsupported operand type'):
+            st + 4
```

Comment:
Why doesn't this work? I thought we enabled it by handling scalars in the binary ops?